docs/sidechain/spec.md

```diff
@@ -0,0 +1,945 @@
+meant for cross-chain transfers. A special set of "witness servers" are used to
+prove that assets were put into trust on the mainchain or returned on the
+sidechain. A new ledger object called a "cross-chain sequence number" is used to
+prevent transaction replay.
+
+A source chain is the chain where the cross-chain transfer start - by either
+putting assets into trust on the mainchain or returning wrapped assets on the
+sidechain. The steps for moving funds from a source chain to a destination chain
+are:
+
+1) On the destination chain, an account submits a transaction that adds a ledger
+   object that will be used to identify the initiating transaction and prevent
+   the initiating transaction from being claimed on the destination chain more
+   than once. The door account will keep a new ledger object - a sidechain. This
+   sidechain ledger object will keep a counter that is used for "cross chain
+   sequence numbers". A "cross chain sequence number" will be checked out from
+   this counter and the counter will be incremented. Once checked out, the
+   sequence number would be owned by the account that submitted the transaction.
+   See the section below for what fields are present in the new sidechain ledger
+   object and cross chain sequence number ledger object. The actual number must
+   be retrieved from the transaction metadata on a validated ledger.
+   
+2) On the source chain, an initiating transaction is sent from a source account.
+   This transaction will include the sidechain, "cross chain sequence number"
+   from step (1), a signature reward amount, in XRP, and an optional destination
+   account on the destination chain. Rewards amounts much match the amount
+   specified by the sidechain ledger object. Both the asset being transferred
+   cross-chain and the reward amount will be transferred from the source account
+   to the door account. Collecting rewards is discussed below. This transaction
+   will create a `XChainTransferRewardState` ledger object for this transaction.
+   It will be owned by the door account.
+   
+3) When a witness servers sees a new cross-chain transaction, it submits a
+   transaction on the destination chain that adds a signature witnessing the
+   cross-chain transaction. This will include the amount being transferred
+   cross-chain, the reward amount, the account to send the reward to, and the
+   optional destination account. These signatures will be accumulated on the
+   cross-chain sequence number object. The keys used in these signatures must
+   match the keys on the multi-signers list on the door account.
+
+4) When a quorum of signatures have been collected, the cross-chain funds can be
+   claimed on the destination chain. If a destination account is specified, the
+   funds will move when the transaction that adds the last signature is executed
+   (the signature that made the quorum). On success, the cross-chain sequence
```

Comment:
> If a destination account is specified, the funds will move when the transaction that adds the last signature is executed (the signature that made the quorum).
This feels semi-trusted, but I'll hold judgment until I get further into this spec.